invisible_cities/reco/deconv_functions_test.py

```diff
@@ -25,12 +29,55 @@
 from .. io      .dst_io           import load_dst
 
 from .. types   .symbols          import InterpolationMethod
+from .. types   .symbols          import CutType
 
 from .. database.load_db          import DataSiPM
 
 from scipy.stats                  import multivariate_normal
 
 
+cut_types = list(CutType)
+
+@pytest.fixture(scope='session')
+def sat_arr(ICDATADIR):
+    '''
+    An array made to imitate a z-slice that would be passed through deconvolution
+    '''
+    arr = np.array([[1.   , 1.   , 0.3  , 0.1  , 1.   ],
+                    [1.   , 1.   , 0.1  , 0.1  , 0.1  ],
+                    [0.1  , 0.1  , 0.1  , 0.1  , 0.1  ],
+                    [0.2  , 0.1  , 1.   , 1.   , 1.   ],
+                    [0.1  , 0.1  , 1.   , 1.   , 1.   ]])
+    return arr
+
+@pytest.fixture(scope='session')
+def compsize_array(ICDATADIR):
+    '''
+    An array made to imitate a z-slice that would be passed through deconvolution
+    This array has specific values such that differing e_cut reveals boolean arrays
+    with differing components across the array.
+
+    Described in more detail in test_component_sizes()
+    '''
+    arr = np.array([[0.5, 0.3, 0.1, 0.3],
+                    [0.3, 0.5, 0.1, 0.1],
+                    [0.1, 0.3, 0.5, 0.1],
+                    [0.1, 0.1, 0.1, 0.5]])
+    return arr
+
+@pytest.fixture(scope='session')
+def no_satellite_killer(ICDATADIR):
+    '''
+    The default satellite_killer parameters provided if you don't want it to be run on
+    your sample. Used to avoid its usage in certain tests.
+    '''
+    satellite_params = dict(satellite_iter     = 9999, 
```

Comment:
I'm thinking that `satellie_iter = np.inf` would be more informative.
Allowing `None` would also be nice.
And calling it `satellite_start_iter`, as `satellite_iter` might be slightly ambiguous